hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/datetime/HazelcastExtractFunction.java

```diff
@@ -71,4 +81,31 @@ public void unparse(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec)
 
         writer.endFunCall(frame);
     }
+
+    private static boolean isSourceValid(RelDataType sourceType) {
```

Comment:
If I understand correctly it does not fail here if we try to for instance extract `CENTURY` from `TIME`, instead such extraction fails in runtime? Could not we move that validation earlier and fail faster?

Reply:
I have tried it but it doesn't work with parameters since at this point we only know that it is Object(ANY).